app/controllers/repositories_controller.rb

```diff
@@ -1,11 +1,64 @@
 class RepositoriesController < ApplicationController
   before_action :load_vars
   before_action :check_view_all_permissions, only: :index
+  before_action :check_edit_permissions, only: %(destroy destroy_modal
+                                                 rename_modal update)
```

Comment:
There should be different before_action for destroy permissions :check_destroy_permissions